app/android/src/uk/co/lutraconsulting/MMAndroidPosition.java

```diff
@@ -0,0 +1,190 @@
+            }
+        };
+
+        return new MMAndroidPosition(context, callback, useFused);
+    }
+
+    private final Context mContext;
+    private final LocationManager mLocationManager;
+    private final boolean mUseFused;
+    private FusedLocationProviderClient mFusedLocationClient = null;
+    private final LocationCallback mLocationCallback;
+    private final LocationListener mLocationManagerCallback;
+    private final GnssStatus.Callback mGnssStatusCallback;
+    private final MMAndroidPosition.Callback mClientCallback;
+    private boolean mFusedAvailable = false;
+    private boolean mGpsProviderAvailable = false;
+    private boolean mIsStarted = false;
+    private String mErrorMessage;
+    private GnssStatus mLastGnssStatus;
+
+    public MMAndroidPosition(Context context, MMAndroidPosition.Callback clientCallback, boolean useFused) {
+        mContext = context;
+        mClientCallback = clientCallback;
+        mUseFused = useFused;
+
+        Log.i("CPP", "[java] constructor!");
+
+        mLocationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+
+        if (mUseFused) {
+            GoogleApiAvailability googleApiAvailability = GoogleApiAvailability.getInstance();
+            mFusedAvailable = googleApiAvailability.isGooglePlayServicesAvailable(context) == ConnectionResult.SUCCESS;
+            Log.i("CPP", "[java] fused available: " + mFusedAvailable);
+            if (mFusedAvailable) {
+                mFusedLocationClient = LocationServices.getFusedLocationProviderClient(context);
+            }
+        } else {
+            mGpsProviderAvailable = mLocationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+            Log.i("CPP", "[java] gps provider available: " + mGpsProviderAvailable);
+        }
+
+        mLocationCallback = new LocationCallback() {
+            @Override
+            public void onLocationResult(@NonNull LocationResult locationResult) {
+                for (Location location : locationResult.getLocations()) {
+                    Log.i("CPP", "[java] FLP " + location.getLatitude() + " " + location.getLongitude());
+
+                    // call the native function!
+                    mClientCallback.onPositionChanged(location, mLastGnssStatus);
+                }
+            }
+        };
+
+        mGnssStatusCallback = new GnssStatus.Callback() {
+            @Override
+            public void onSatelliteStatusChanged(@NonNull GnssStatus status) {
+                //Log.v("LOC", "GNSS Status: " + status.getSatelliteCount() + " satellites.");
+
+                // store the satellite info
+                mLastGnssStatus = status;
+            }
+        };
+
+        mLocationManagerCallback = new LocationListener() {
+            @Override
+            public void onLocationChanged(@NonNull Location location) {
+                Log.i("CPP", "[java] GPS " + location.getLatitude() + " " + location.getLongitude());
```

Comment:
I see a bunch of logs, do you want to keep them there?

Reply:
I would keep them, at least for now until we fine tune this :slightly_smiling_face: